generator/processors/Microsoft.Resources.ts

```diff
@@ -24,8 +75,13 @@ function appendProps(definition: any, props: any[]) {
 }
 
 export const postProcessor: SchemaPostProcessor = (namespace: string, apiVersion: string, schema: any) => {
-  appendProps(schema?.tenant_resourceDefinitions?.deployments, [subscriptionProps]);
-  appendProps(schema?.managementGroup_resourceDefinitions?.deployments, [subscriptionProps]);
-  appendProps(schema?.subscription_resourceDefinitions?.deployments, [resourceGroupProps]);
-  appendProps(schema?.resourceDefinitions?.deployments, [subscriptionProps, resourceGroupProps]);
+  appendProps(schema.tenant_resourceDefinitions?.deployments, [subscriptionProps]);
+  appendProps(schema.managementGroup_resourceDefinitions?.deployments, [subscriptionProps]);
+  appendProps(schema.subscription_resourceDefinitions?.deployments, [resourceGroupProps]);
+  appendProps(schema.resourceDefinitions?.deployments, [subscriptionProps, resourceGroupProps]);
+
+  if (lowerCaseCompare(apiVersion, '2018-05-01') > -1) {
+    schema.subscription_resourceDefinitions = schema.subscription_resourceDefinitions ?? {};
```

Comment:
do we need this line?

Reply:
Yes - the schema output may not necessarily contain a `subscription_resourceDefinitions` section if there are no subscription-level resources defined